src/ShopBundle/objects/db/TShopArticle.class.php

```diff
@@ -408,6 +410,8 @@ protected function GetDetailLinkV2($bIncludePortalLink = false, $iCategoryId = n
         $oManufacturer = $this->GetFieldShopManufacturer();
         if (!is_null($oManufacturer)) {
             $aParts[] = $urlNormalizationUtil->normalizeUrl($oManufacturer->fieldName);
+        } else {
```

Comment:
else in combination with negative comparisons is hard to read - could you switch to a positive comparison and switch if/else?